code/game/objects/items/storage/boxes.dm

```diff
@@ -609,6 +609,43 @@
 	playsound(loc, "rustle", 50, 1, -5)
 	user.visible_message("<span class='notice'>[user] hugs \the [src].</span>","<span class='notice'>You hug \the [src].</span>")
 
+/////clown box & honkbot assembly
+obj/item/storage/box/clown
+	name = "clown box"
+	desc = "A colorful cardboard box for the clown"
+	icon_state = "clownbox"
+	illustration = null
+	//foldable = null
+
+/obj/item/stack/sheet/cardboard/attackby(obj/item/stamp/clown/I, mob/user, params)
+	if(!istype(I, /obj/item/stamp/clown))
+		..()
+		return
+	to_chat(user, "<span class='notice'>You stamp the cardboard! Its a clown box! Honk!</span>")
+	var/turf/T = get_turf(src)
+	var/obj/item/storage/box/clown/A = new /obj/item/storage/box/clown(T)
+	user.put_in_hands(A)
+	qdel(src)
+
+/obj/item/storage/box/clown/attackby(obj/item/I, mob/user, params)
+	if((istype(I, /obj/item/bodypart/l_arm/robot)) || (istype(I, /obj/item/bodypart/r_arm/robot)))
+		if(contents.len >= 1) //prevent accidently deleting contents

```

Comment:
you only need `if(contents.len)` here

Reply:
you do. without it, you'll create the assembly by deleting everything inside the box.

Reply:
You might be confusing with just `contents`, what I said already implies it will be 1 or more(since 0 would not trigger this if and can't have list length < 0 )

Reply:
I'll test it without, then.

Reply:
Yeah, nah, without the .len - it won't even build an assembly, nor show the message.
We need it.

Reply:
We're not talking about removing the `.len`, we're talking about removing the `>=1`

Reply:
Oh!!